src/ucx_plugin.c

```diff
@@ -236,12 +235,20 @@ static ncclResult_t ucx_init_context(ucp_context_h *ctx, int dev) {
 
 static ncclResult_t ucx_init_worker(ucp_context_h ctx, ucp_worker_h *worker) {
   ucp_worker_params_t worker_params;
+  ucp_worker_attr_t   worker_attr;
 
   memset(&worker_params, 0, sizeof(worker_params));
   worker_params.field_mask  = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
-  worker_params.thread_mode = UCS_THREAD_MODE_SINGLE;
+  worker_params.thread_mode = UCS_THREAD_MODE_MULTI;
```

Comment:
does it have to be the default.  is it possible to check if nccl is running multi-thread mode and set it accordingly?

Reply:
I don't think we can know in advance how many threads will be used, at least at worker creation time.